### ISSUE TITLE
Move / copy media doesn't retain custom properties

### ISSUE THREAD

**Issue by mradamwood** [spatie/laravel-medialibrary#1073]
When moving images between models the custom properties I've set aren't being moved.

I updated the copy function  by removing

`$newMedia->custom_properties = $this->custom_properties;`

and adding 

` ->withCustomProperties($this->custom_properties)`

to make the function

```php
public function copy(HasMedia $model, $collectionName = 'default'): self
    {
        $temporaryDirectory = TemporaryDirectory::create();

        $temporaryFile = $temporaryDirectory->path($this->file_name);

        app(Filesystem::class)->copyFromMediaLibrary($this, $temporaryFile);

        $newMedia = $model
            ->addMedia($temporaryFile)
            ->usingName($this->name)
            ->withCustomProperties($this->custom_properties)
            ->toMediaCollection($collectionName);

        $temporaryDirectory->delete();

        return $newMedia;
    }
```

which solves the problem for me. Using 7.3.0.

**Comment by brendt** [comment]
@mradamwood are you sure that there's no other reason for your custom properties not te be moved in your code? There's a test in `\Spatie\MediaLibrary\Tests\Feature\Models\Media\CopyTest` which tests for customer properties to be copied to, and it works, even with the old implementation.

**Comment by mradamwood** [comment]
The copy test at `\Spatie\MediaLibrary\Tests\Feature\Models\Media\CopyTest` uses withCustomProperties()

```php
$media = $model
     ->addMedia($this->getTestJpg())
     ->usingName('custom-name')
     ->withCustomProperties(['custom-property-name' => 'custom-property-value'])
     ->toMediaCollection();
```

**Comment by brendt** [comment]
@mradamwood That's right, but the actual copying still is the same?

```php
$movedMedia = $media->copy($anotherModel, 'images');
```

Are you able to make a failing test with the latest version of the media library? Because it seems that I'm not able to reproduce it.

**Comment by mradamwood** [comment]
@brendt my test passed when I based it on yours but when I added in another test to get the custom property via the new model, not directly from `$movedMedia` the media, it fails.

```php
$this->assertEquals($anotherModel->getMedia('images')->first()->getCustomProperty('custom-property-name'),
```

```
1) Tests\Feature\CopyTest::it_can_copy_media_from_one_model_to_another
Failed asserting that 'custom-property-value' matches expected null.
```
You can also see from below that the custom properties aren't saved to the database.

<img width="429" alt="screen shot 2018-05-02 at 09 57 53" src="https://user-images.githubusercontent.com/1739469/39514595-89586dd4-4def-11e8-9e7e-44aa2ccc1159.png">

Looks like the info is stored in `$movedMedia` but doesn't save to database.

**Comment by brendt** [comment]
Right, got it. I'm working on a fix :)

**Comment by brendt** [comment]
This is fixed as of 7.3.1: https://github.com/spatie/laravel-medialibrary/releases/tag/7.3.1

**Comment by Hansen6171** [comment]
@brendt  how to use moving spatie media library in documentation ?
